### PR TITLE
Add JWT Support

### PR DESCRIPTION
**Pull request by qreqit**

### CODE REVIEW

File: src/main/java/mate/academy/springbootwebgreqit/controller/AuthenticationController.java [hunk 1]
```diff
@@ -11,10 +14,16 @@
 @RequiredArgsConstructor
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto registerUser(@RequestBody UserRegistrationRequestDto requestBody) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/controller/AuthenticationController.java:20]:
```suggestion
    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto requestBody) {
```

File: src/main/java/mate/academy/springbootwebgreqit/controller/AuthenticationController.java [hunk 2]
```diff
@@ -11,10 +14,16 @@
...
         return userService.register(requestBody);
     }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto loginUser(@RequestBody UserLoginRequestDto response) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/controller/AuthenticationController.java:25]:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto response) {
```

File: src/main/java/mate/academy/springbootwebgreqit/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/security/JwtAuthenticationFilter.java:46]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/mate/academy/springbootwebgreqit/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/security/JwtAuthenticationFilter.java:47]:
```suggestion
        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_HEADER)) {
```

create constant for "Bearer "

File: src/main/java/mate/academy/springbootwebgreqit/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/security/JwtAuthenticationFilter.java:48]:
```suggestion
            return bearerToken.substring(TOKEN_HEADER.length());
```

File: src/main/java/mate/academy/springbootwebgreqit/dto/user/UserLoginResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.springbootwebgreqit.dto.user;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@RequiredArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/dto/user/UserLoginResponseDto.java:7]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/springbootwebgreqit/security/JwtUtil.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+    private <T> T getClaimsFromToken(String token, Function<Claims, T> claimsResolver) {
+        final  Claims claims = Jwts.parserBuilder()
+                .setSigningKey(secret)
+                .build()
+                .parseClaimsJws(token)
+                .getBody();
+        return claimsResolver.apply(claims);
+    }
+}
```
Comment (okuzan) [inline src/main/java/mate/academy/springbootwebgreqit/security/JwtUtil.java:59]:
```suggestion
}

```

**Review by okuzan — APPROVED** [review]

**Pull request merged**